components/list/list-item-drag-mixin.js

```diff
@@ -33,86 +95,299 @@ export const ListItemDragMixin = superclass => class extends superclass {
+			const dropSpots = dropSpotsFactory();
+			dropSpots.setDestination(this, move.below);
+			this._inTopArea = false;
+		}
+	}
+
+	_drop() {
+		const dropSpots = dropSpotsFactory();
+		dropSpots.setDestination(this, dropSpots.moveBeforeDestination);
+	}
+
+	_hostDragEnter(e) {
+		const dropSpots = dropSpotsFactory();
+		if (this === dropSpots.target) {
+			return;
+		}
+		dropSpots.addVisitor(this);
+		this._dropTarget = true;
+		e.dataTransfer.dropEffect = 'move';
+	}
+
+	_moveItem(targetKey, destinationKey, moveBeforeDestination = false, temporaryMovement = false) {
 		this.dispatchEvent(new CustomEvent('d2l-list-item-position', {
-			detail: detail,
+			detail: new NewPositionEventDetails({targetKey, destinationKey, moveBeforeDestination, temporaryMovement}),
 			bubbles: true
 		}));
 	}
 
-	_dragEnter() {
-		this._dragStartHandler();
+	_renderBottomPlacementMarker(renderTemplate) {
+		return this._bottomPlacementMarker ? html`<div class="d2l-list-item-drag-bottom-marker">${renderTemplate}</div>` : null;
+	}
+
+	_renderDraggableArea(templateMethod) {
+		templateMethod = templateMethod || (dragArea => dragArea);
+		return this.draggable && !this._keyboardMode ? templateMethod(html`
+			<div
+				class="d2l-list-item-drag-area"
+				draggable="true"
+				@click="${this._dragAreaClick}"
+				@dragstart="${this._dragStart}"
+				@dragend="${this._dragEnd}"
+				>
+			</div>
+		`) : nothing;
+	}
+
+	_renderDragHandle(templateMethod) {
+		templateMethod = templateMethod || (dragHandle => dragHandle);
+		return this.draggable ? templateMethod(html`
+			<d2l-list-item-drag-handle id="${this._itemDragId}" @d2l-list-item-drag-handle-action="${this._dragHandleActions}"></d2l-list-item-drag-handle>
+		`) : nothing;
 	}
 
-	_dragExit() {
-		this._dispatchDragEvent(null);
+	_renderDropArea(templateMethod) {
+		templateMethod = templateMethod || (dropArea => dropArea);
+		return this.draggable && this._dropTarget ? templateMethod(html`
+			<div class="d2l-list-item-drag-drop-grid" @drop="${this._drop}" @dragover="${this._dragOver}">
+				<div @dragenter="${this._dragTopEnter}"></div>
+				<div @dragenter="${this._dragUpperEnter}"></div>
+				<div @dragenter="${this._dragLowerEnter}"></div>
+				<div @dragenter="${this._dragBottomEnter}"></div>
+			</div>
+		`) : nothing;
 	}
 
-	_dragStartHandler() {
-		this.moveBefore(null);
-		this._dispatchDragEvent(null);
+	_renderTopPlacementMarker(renderTemplate) {
+		return this._topPlacementMarker ? html`<div class="d2l-list-item-drag-top-marker">${renderTemplate}</div>` : null;
+	}
+};
+
+let dropSpots = null;
+
+function dropSpotsFactory(target) {
+	if (!dropSpots) dropSpots = new keepTrackOfThoseDropSpots(target);
+	return dropSpots;
+
+}
+
+function dropSpotsBlowUp() {
+	const dropSpotTemp = dropSpots;
+	if (dropSpots) {
+		dropSpots.clear();
 	}
+	dropSpots = null;
+	return dropSpotTemp;
+}
 
-	_dragStopHandler() {
-		this._dispatchDragEvent(null);
-		this.moveafter(null);
+class keepTrackOfThoseDropSpots {
```

Comment:
Classes should always be capitalized - this can also probably be simplified to `DropSpotsState` since it is in effect a state object

Reply:
Yay, thank you I really hated the name :D